pytorch_lightning/core/optimizer.py

```diff
@@ -298,10 +301,13 @@ def _configure_schedulers_automatic_opt(schedulers: list, monitor: Optional[str]
             lr_schedulers.append(
                 {**default_config, "scheduler": scheduler, "reduce_on_plateau": True, "monitor": monitor}
             )
-        elif isinstance(scheduler, optim.lr_scheduler._LRScheduler):
-            lr_schedulers.append({**default_config, "scheduler": scheduler})
         else:
-            raise ValueError(f'The provided lr scheduler "{scheduler}" is invalid')
+            lr_schedulers.append({**default_config, "scheduler": scheduler})
+
+        current_scheduler = lr_schedulers[-1]["scheduler"]
+        if not isinstance(current_scheduler, _SupportedLRScheduler):
+            raise ValueError(f"The provided lr scheduler `{current_scheduler.__class__.__name__}` is invalid.")
```

Comment:
I would remove this personally.

We had a ValueError there before to fill in the `else` condition, but it's not necessary anymore. Also checking the last one is a bit arbitrary.

Reply:
the first condition `if isinstance(scheduler, dict):` doesn't check for the scheduler protocol API in its block and the else condition we had won't be triggered if the user puts in an invalid scheduler even though all the keys are present. This avoids duplicate assertions and checks the recently added scheduler or not at one place.

Reply:
Still, we don't *need* to check this here. This function does not call any of the methods defined by "_SupportedLRScheduler".

Additionally, we don't *need* the scheduler to define this interface as the user can override the actually calls and do `scheduler.foo()` instead of `scheduler.step()`

Reply:
that is something @tchaton and @ananthsub proposed to have certain boundaries, for now, to make sure that basic API should be followed while implementing the even a custom scheduler and at the lowest level it should have a step, state_dict and load_state_dict methods defined else such a scheduler won't work within lightning.

Reply:
If the matter is with state_dict, then we can use the existing `_SupportsStateDict` to check it. But it should be in the checkpoint connector IMO.

And as I mentioned in my previous comment, we don't need to enforce `step()` anymore since this PR allows customization.